### PR TITLE
Fixes for 1.0

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

File: src/OutputCollector.jl [hunk 1]
```diff
@@ -157,12 +151,12 @@ function wait(collector::OutputCollector)
...
         wait(collector.P)
-        wait(collector.stdout_linestream.task)
-        wait(collector.stderr_linestream.task)
+        fetch(collector.stdout_linestream.task)
+        fetch(collector.stderr_linestream.task)
 
         # Also fetch on any extra tasks we've jimmied onto the end of this guy
         for t in collector.extra_tasks
-            wait(t)
+            fetch(t)
```
Comment (quinnj) [inline src/OutputCollector.jl]: @Keno @staticfloat, while trying to do `build SpecialFunctions` on 1.0 (osx), I'm seeing BinaryProvider hang on this line when trying to execute the `curl` command to download the binary. I'm not sure what's going on.

File: src/PlatformEngines.jl [hunk 2]
```diff
@@ -451,7 +451,9 @@ function download(url::AbstractString, dest::AbstractString;
         if isa(e, InterruptException)
             rethrow()
         end
-        error("Could not download $(url) to $(dest)")
+        st = stacktrace(catch_backtrace())
+        @error "Could not download $(url) to $(dest)" exception=(e, st)
```
Comment (ararslan) [inline src/PlatformEngines.jl]: Why is this necessary?

Comment (quinnj) [reply src/PlatformEngines.jl]: Not strictly necessary, but it enabled me to actually see what error was being thrown instead of swallowing it in a generic `error` call.

**Comment by staticfloat** [comment]
I started a branch based off of yours, then Keno started a branch based off of mine.  All this to say, closing in favor of #95.  Also, I have not been able to reproduce  the freeze on downloading `SpecialFunctions`.  I installed it just fine today.  :(
